src/NHibernate/Persister/Entity/IPropertyMapping.cs

```diff
@@ -29,6 +29,8 @@ public interface IPropertyMapping
 		/// <returns>true if a type was found, false if not</returns>
 		bool TryToType(string propertyName, out IType type);
 
+		string[] ToColumns(ICriteria pathCriteria, string propertyName, System.Func<ICriteria, string> getSQLAlias);
```

Comment:
I didn't look in to actual implementation but this change looks wrong. `IPropertyMapping` interface shouldn't depend on any query API (`ICriteria` in your case)